test/tests/TileElements.cpp

```diff
@@ -140,3 +147,28 @@ TEST_F(TileElementWantsFootpathConnection, DifferentHeight)
     EXPECT_FALSE(tile_element_wants_path_connection_towards({ 18, 10, 24, 1 }, nullptr));
     SUCCEED();
 }
+
+TEST_F(TileElementWantsFootpathConnection, MapEdge)
+{
+    // Paths at the map edge should have edge flags turned on when placed in scenario editor
+    TileElement* pathElement = map_get_footpath_element(TileCoordsXYZ{ 1, 4, 14 }.ToCoordsXYZ());
+
+    // Enable the scenario editor flag
+    gScreenFlags |= SCREEN_FLAGS_SCENARIO_EDITOR;
+
+    // Calculate the connected edges
+    footpath_connect_edges({ 16, 64 }, pathElement, 0);
+    auto edges = pathElement->AsPath()->GetEdges();
+
+    // The tiles alongside in the Y direction are both on the map edge so should be marked as an edge
+    EXPECT_TRUE(edges & (1 << 1));
+    EXPECT_TRUE(edges & (1 << 3));
```

Comment:
Thought about this a bit, and I'm not sure this is desired behaviour. The `map_is_edge` call at Footpath.cpp:834 will return true if the coordinates are for a piece touching the edge of the map or off the edge of the map. This means a footpath piece will try to connect to empty tiles on the edge of the map, which I don't think is correct. I can't actually think of how this could break anything, though.